fish.pc.in

```diff
@@ -1,8 +1,8 @@
 prefix=@prefix@
-datadir=@datadir@
-completionsdir=@extra_completionsdir@
-functionsdir=@extra_functionsdir@
-confdir=@extra_confdir@
+datadir=${prefix}/@rel_datadir@
+completionsdir=${datadir}/@rel_completionsdir@
```

Comment:
So, this will just not work because, while the completionsdir will now be set in the pkg-config file, and so the completions will be *installed* there, fish won't *read* them because __fish_build_paths will still contain the paths in /usr/local.

Reply:
This patch doesn't change the completions installation path, nor the paths where fish searches completions.

Reply:
Yes, but it changes where things would be installed via pkg-config.

Reply:
Without altering where fish *looks* - that's what `__fish_build_paths.fish` contains.

Reply:
But fish already searches in `/usr/share/fish/vendor_completions.d`, right? `extra_completionsdir` is just an _extra_ directory.

Reply:
No, vendor_completions.d *is* extra_completionsdir.

Reply:
Before this patch:

* System completions are installed in `/usr/share/fish/vendor_completions.d`
* They are looked up by fish
* `/usr/local/share/fish/vendor_completions.d` is used in the pkg-config file but doesn't exist
* `extra_completionsdir` is set to `/usr/local/share/fish/vendor_completions.d`

```
> ls /usr/share/fish/vendor_completions.d
alacritty.fish  curl.fish  hub.fish  pass.fish  rg.fish  sway.fish  swayidle.fish  swaylock.fish  swaymsg.fish  swaynag.fish  yay.fish  youtube-dl.fish
> ls /usr/local/share/fish/vendor_completions.d
ls: cannot access '/usr/local/share/fish/vendor_completions.d': No such file or directory
> cat /usr/share/pkgconfig/fish.pc 
…
completionsdir=/usr/local/share/fish/vendor_completions.d
…
```

Reply:
Why are the completions installed in /usr/share/fish/vendor_completions.d if the pkg-config file says /usr/local/share/fish/vendor_completions.d?

The reason this also looks up /usr/share/fish/vendor_completions.d is because of $XDG_DATA_DIRS containing /usr/share, so we add that to the searchpath.

But what *should* be changed is that extra_completionsdir should typically point to /usr/share/fish/vendor_completions.d, with "/usr" being the prefix.

Currently, it will change the pkg-config entry but not the fish variable, so if the prefix is set to something that *isn't* in $XDG_DATA_DIRS the directory won't be searched.

Reply:
>Why are the completions installed in /usr/share/fish/vendor_completions.d if the pkg-config file says /usr/local/share/fish/vendor_completions.d?

Because currently no-one uses pkg-config to find where completions should be installed. And it's a good thing since doing so would install to `/usr/local`, which is definitely not something a distro package should do.

>The reason this also looks up /usr/share/fish/vendor_completions.d is because of $XDG_DATA_DIRS containing /usr/share, so we add that to the searchpath.

Oh, I see. Okay, let's change that too. People will still be able to override it via build options anyway.